controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -727,9 +717,16 @@ public OperationContext createContext(String scope, String name) {
     }
 
     @Override
-    public CompletableFuture<Void> completeCommitTransactions(String scope, String stream, VersionedMetadata<CommittingTransactionsRecord> record,
+    public CompletableFuture<Void> completeCommitTransactions(String scope, String streamName, VersionedMetadata<CommittingTransactionsRecord> record,
                                                               OperationContext context, ScheduledExecutorService executor) {
-        return withCompletion(getStream(scope, stream, context).completeCommittingTransactions(record), executor);
+        Stream stream = getStream(scope, streamName, context);
+        CompletableFuture<Void> future = withCompletion(stream.completeCommittingTransactions(record), executor);
+
+        return future.thenCompose(result -> {
+            return stream.getNumberOfOngoingTransactions().thenAccept(count -> {
```

Comment:
Regarding the comment about this operation being expensive, I see that for ZK, it is implemented like this:

```
@Override
    public CompletableFuture<Integer> getNumberOfOngoingTransactions() {
        return store.getChildren(activeTxRoot).thenCompose(list ->
                Futures.allOfWithResults(list.stream().map(epoch ->
                        getNumberOfOngoingTransactions(Integer.parseInt(epoch))).collect(Collectors.toList())))
                    .thenApply(list -> list.stream().reduce(0, Integer::sum));
    }
```

Perhaps we could implement this method by simply taking the number of children from the `activeTxRoot` from the `stat`rather than listing the children, which is more expensive. If it makes sense, then let's not do it here, but create a new issue to work on it. cc/ @shiveshr

Reply:
Done, see https://github.com/pravega/pravega/issues/3216.